app/master/build.py

```diff
@@ -61,12 +62,15 @@ def __init__(self, build_request):
         self._postbuild_tasks_are_finished = False  # WIP(joey): Remove and use build state.
         self._timing_file_path = None
 
+        leave_state_callbacks = {build_state: self._on_leave_state
+                                 for build_state in BuildState}
```

Comment:
It might be simpler to roll this into `onchangestate` instead. (That's where the absolute timestamp values are already being set so it would make sense to record deltas there as well.)

From [Fysom docs](https://github.com/mriehl/fysom#callbacks):
> In addition, a generic onchangestate() callback can be used to call a single function for all state changes.

It looks like if you do that you also might not need to do an additional `time.time()` call in the callback since at that point the `transition_timestamps` value for the new state would already be set. That's not a huge deal though.